### PR TITLE
End-to-end test

### PR DESCRIPTION
**Pull request by yshdb5**
### Description:
Implemented the infrastructure that allows end-to-end testing with Kaspresso. This allows to test the entire seeker's UI view of the application and its overall functionality.

### Related Issues (if applicable):
- Fixes issue #141 

### Changes Made:
- Added the kaspresso.screens package with kaspresso objects.
- Added the kaspresso.screenTests package with kaspresso tests for each screen.
- Added the kaspresso.end2end package with the final end-to-end test file which still needs to be implemented.
- Moved ProviderRegistrationTest file to its correct package.

### How to Test:
- Simply run kaspresso.screenTests file in AndroidTests or the connectedCheck

### CODE REVIEW

**Review by ethanboren — APPROVED** [review]
Great work on the end-to-end tests! These were absolutely necessary to ensure everything is functioning correctly and that our work is solid. Your thoroughness and attention to detail have helped identify key areas and strengthen the reliability of our project.

Thanks for putting in such essential and high-quality effort!

File: app/src/androidTest/java/com/android/solvit/kaspresso/end2end/EndToEndTest.kt
Comment (achillet3) [inline app/src/androidTest/java/com/android/solvit/kaspresso/end2end/EndToEndTest.kt:1]: What is the purpose of this file?

File: app/src/androidTest/java/com/android/solvit/kaspresso/screenTests/ChooseProfileScreenTest.kt
Comment (achillet3) [inline app/src/androidTest/java/com/android/solvit/kaspresso/screenTests/ChooseProfileScreenTest.kt:1]: ## Overall Comments
- Good implementation of the tests for this step
- Does not verify that each action has been correctly performed
- - Example: "Perform the Choose Seeker profile action" test could check if the navigation has been implemented correctly

File: app/src/androidTest/java/com/android/solvit/kaspresso/screenTests/ChooseProfileScreenTest.kt [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        providerButton.assertIsDisplayed()
+        learnMoreLink.assertIsDisplayed()
+      }
+    }
+
+    step("Perform the Choose Seeker profile action") {
+      // Perform the Choose Profile action
+      ComposeScreen.onComposeScreen<ChooseProfileScreenObject>(composeTestRule) {
+        customerButton.performClick()
+      }
```
Comment (achillet3) [inline app/src/androidTest/java/com/android/solvit/kaspresso/screenTests/ChooseProfileScreenTest.kt:45]:
```suggestion
      // Verify that navigation occurred
  verify {
    val navigationActions = mock(NavigationActions::class.java)
    verify(navigationActions).navigateToCustomerOnboarding()
  }
      }
```

**Review by achillet3 — APPROVED** [review]
## Overall Comments
- Good implementation of the tests with Kaspresso
- Missing real end-to-end tests to verify an entire user flow while checking that navigation between screens is implemented correctly

## Potential Improvements
- Could add tests to verify the actions have been performed correctly (see ChooseProfileScreenTest.kt file comment)
- Could add negative tests to check that the error handling is implemented correctly
- Implement tests to check navigation between screens
-

**Pull request merged**